dev-lang/rust: respect *FLAGS for USE=-system-llvm build

matoro opened the pull request.

gyakovlev commented:
I'm curious, does it override or appends flags to the build?
haven't tested it yet ofc.

matoro commented:
> I'm curious, does it override or appends flags to the build? haven't tested it yet ofc.

It's append, see [docs](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG.html#variable:CMAKE_%3CLANG%3E_FLAGS_%3CCONFIG%3E).

By default, `CMAKE_{C,CXX}_FLAGS` are supposed to be taken from the environment, but LLVM clobbers those with the ones it needs.  However it doesn't touch `CMAKE_{C,CXX}_FLAGS_{DEBUG,RELEASE}` and leaves them at the cmake defaults (which includes -O3 for RELEASE), hence why we can safely stick user CFLAGS there.

Actually, I just tried out USE=debug, and it uses the DEBUG cmake configuration for the bundled LLVM.  So I need to adjust that too.

gyakovlev commented:
got a failure
```
[106/2071] Linking CXX static library lib/libLLVMDemangle.a                                                                                                                                                                                                     
FAILED: lib/libLLVMDemangle.a                                                                                                                                                                                                                                   
: && /usr/bin/cmake -E rm -f lib/libLLVMDemangle.a && /usr/sbin/powerpc64le-unknown-linux-gnu-ar Dqc lib/libLLVMDemangle.a -O2 -mcpu=power9 -ftree-vectorize -pipe -frecord-gcc-switches -fdiagnostics-show-option -fdiagnostics-color=always -Wl,-O1 -Wl,--as-n
eeded -Wl,-z,relro,-z,now -Wl,--defsym=__gentoo_check_ldflags__=0 lib/Demangle/CMakeFiles/LLVMDemangle.dir/Demangle.cpp.o lib/Demangle/CMakeFiles/LLVMDemangle.dir/ItaniumDemangle.cpp.o lib/Demangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangle.cpp.o lib/De
mangle/CMakeFiles/LLVMDemangle.dir/MicrosoftDemangleNodes.cpp.o lib/Demangle/CMakeFiles/LLVMDemangle.dir/RustDemangle.cpp.o lib/Demangle/CMakeFiles/LLVMDemangle.dir/DLangDemangle.cpp.o && /usr/sbin/powerpc64le-unknown-linux-gnu-ranlib -D lib/libLLVMDemangl
e.a && :                                                                                                                                                                                                                                                        
/usr/sbin/powerpc64le-unknown-linux-gnu-ar: invalid option -- '2' 
```

```
portageq envvar LDFLAGS
-O2 -mcpu=power9 -ftree-vectorize -pipe -frecord-gcc-switches -fdiagnostics-show-option -fdiagnostics-color=always -Wl,-O1 -Wl,--as-needed -Wl,-z,relro,-z,now -Wl,--defsym=__gentoo_check_ldflags__=0
```

so it's not very smart it seems and can't filter out irrelevant ldflags.
this will introduce funny failures all across the board ( users put a lot of funny stuff in LDFLAGS)

matoro commented:
Converted to revbump so it's tougher to see the changes.  Three diffs from last push:

* Now works with `USE=debug` by setting `CMAKE_{C,CXX}_FLAGS_{DEBUG,RELEASE} appropriately
* LLVM build is now verbose
* Removed static linker flags

gyakovlev commented on dev-lang/rust/rust-1.61.0-r2.ebuild:
```diff
@@ -0,0 +1,731 @@
+		fi
+	fi
+
+	cm_btype="$(usex debug DEBUG RELEASE)"
```
make it `local` please
